### PR TITLE
Added environment variable to store the material changed flag. This w…

### PR DESCRIPTION
**Pull request by jyotisingh**
…ould be useful at the agent side to implement logic that needs the information about the changed materials

### CODE REVIEW

**Comment by arvindsv** [comment]
Yay. Consider: `GO_MATERIAL_%s_HAS_CHANGED`? Documents itself a little better in my opinion, but it is bigger.

**Comment by arvindsv** [comment]
Something is failing. Otherwise, it looks ok to me.

File: common/src/test/java/com/thoughtworks/go/domain/MaterialRevisionsTest.java [hunk 1]
```diff
@@ -487,6 +488,33 @@ public void shouldUseUpstreamPipelineLabelForDependencyMaterial() {
...
+        revisions.addRevision(new MaterialRevision(MaterialsMother.dependencyMaterial("p2", "s2"), false, ModificationsMother.changedDependencyMaterialRevision("p3", 1, "1", "s", 1, new Date()).getModifications()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.pluggableSCMMaterial("scm1", "scm1name"), true, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.pluggableSCMMaterial("scm2", "scm2name"), false, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.packageMaterial("repo1", "repo1name", "pkg1", "pkg1name"), true, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.packageMaterial("repo2", "repo2name", "pkg2", "pkg2name"), false, ModificationsMother.multipleModificationList()));
+
+        revisions.populateEnvironmentVariables(context, null);
+
+        assertThat(context.getProperty("GO_MATERIAL__HAS_CHANGED"), is("true"));
```
Comment (arvindsv) [inline common/src/test/java/com/thoughtworks/go/domain/MaterialRevisionsTest.java]: FYI: Without the change [here](https://github.com/gocd/gocd/pull/4220/files#diff-cdff4287c76672b79e5c4c818b84b689R156), that variable becomes: `GO_MATERIAL_NULL_HAS_CHANGED`.

**Comment by jyotisingh** [comment]
@arvindsv - Your changes look alright to me. Good that you spotted the missing testcase of folder and material-name both being `null`. I added a check for `blank` so that the environment variable that gets generated would be `GO_MATERIAL_HAS_CHANGED` and not `GO_MATERIAL__HAS_CHANGED`, we are good there. Will merge once PR build goes green.

**Pull request merged**